include/swift/AST/Decl.h

```diff
@@ -1957,6 +1957,21 @@ class PatternBindingDecl final : public Decl,
   
   /// Does this binding declare something that requires storage?
   bool hasStorage() const;
+
+  /// Can this binding be default initialized?
+  bool isDefaultInitializable() const {
+    if (!hasStorage())
+      return false;
```

Comment:
I'm not sure this is correct. If the binding doesn't have storage, it doesn't *need* to be default-initialized. That seems closer to a `true` than a `false` for me, but honestly I'd just leave it as an assertion.

Reply:
If whether the binding has storage or not doesn't bear on whether default initialization can happen, does asserting have any value over just taking this check out altogether? Because this was private implementation before the assertion was reasonable, but I don't know whether it makes that much sense now, no?

Reply:
The assertion would indicate that it is a programmer error to call it in a context where you don't already know whether the PBD has storage. It's also questionable whether `true` or `false` would be the right answer in that scenario anyway, so an assertion gets you out of having to answer the question.

Reply:
Fair enough 👍 I'll bring that back